XamarinCommunityToolkit/Views/AvatarView/ImageSourceValidator.android.ios.macos.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+#if MONOANDROID
+using Xamarin.Forms.Platform.Android;
+#elif __IOS__
+using Xamarin.Forms.Platform.iOS;
+#else
+using Xamarin.Forms.Platform.MacOS;
+#endif
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	class ImageSourceValidator : IImageSourceValidator
+	{
+		public async Task<bool> IsImageSourceValid(ImageSource source)
+		{
+			IImageSourceHandler handler;
+
+			switch (source)
```

Comment:
Don't you like the idea of extracting the logic of getting an image handle into a separate method?)

```csharp
IImageSourceHandler GetHandler(ImageSource source)
{
	if (source is UriImageSource)
		return new UriImageSourceHandler();

	if (source is StreamImageSource)
		return new StreamImageSourceHandler();

	if (source is FileImageSource fileSource && File.Exists(f.File))
		return new FileImageSourceHandler()

	return null;
}
```

Reply:
I like the idea putting it into the single platform file. probably I misunderstood you. I thought you suggested to return IImageSourceHandler as a result of IsImageSourceValidAsync(...)

Reply:
Oh, this suggestion is only about code style :)

Replace 

```
IImageSourceHandler handler;
switch (source) 
{
....
}
```

by

```
var handler = GetHandler(source);

IImageSourceHandler GetHandler(ImageSource source)
{
	if (source is UriImageSource)
		return new UriImageSourceHandler();

	if (source is StreamImageSource)
		return new StreamImageSourceHandler();

	if (source is FileImageSource fileSource && File.Exists(f.File))
		return new FileImageSourceHandler()

	return null;
}
```